src/app.js

```diff
@@ -1,196 +1,285 @@
+/**
+ * TODO:
+ * - create all elements dynamically from JS
+ * - leave only a single <div id="root"/> in html
+ * - wrap all the functionality in this module in a single function e.g. "createApp(root)"
+ * - add a new dev.js module where we can use the new `createApp()` function
+ * - call `createApp()` with the root DOM node
+ *
+ * In the test file we will be able to import the same `createApp()` function
+ * to create our application then find elements and interact with it.
+ * */
+
+export default function createApp(rootElement) {
+  const createBtn = (className, innerText) => {
+    const button = document.createElement("button");
+    button.classList.add(className);
+    button.innerHTML = innerText;
+    return button;
+  };
+
+  const createHeading = (headingSize, className, innerText) => {
+    const heading = document.createElement(headingSize);
+    heading.classList.add(className);
+    heading.innerHTML = innerText;
+    return heading;
+  };
+
+  const createHeader = () => {
+    const header = document.createElement("header");
+    const headingOne = createHeading("h1", "header", "Todo App");
+    header.appendChild(headingOne);
+    return header;
+  };
+
+  const createForm = () => {
+    const form = document.createElement("form");
+    return form;
+  };
+
+  const createInput = () => {
+    const inputElement = document.createElement("input");
+    inputElement.type = "text";
+    inputElement.classList.add("new-todo");
+    inputElement.placeholder = "Please enter todo here";
+    return inputElement;
+  };
+
+  const createFilterElement = () => {
+    const selectEl = document.createElement("select");
+    let option;
+    const inputData = "All,completed,Outstanding";
+    inputData.split(",").forEach((item) => {
+      option = document.createElement("option");
+      option.value = item.toLowerCase();
+      option.innerHTML = item;
+      selectEl.appendChild(option);
+    });
+    return selectEl;
+  };
+
+  const createTodoList = () => {
+    const divEl = document.createElement("div");
+    divEl.classList.add("container");
+    const ulEl = document.createElement("ul");
+    ulEl.classList.add("list");
+    divEl.appendChild(ulEl);
+    return divEl;
+  };
+
+  const changeUserBtn = createBtn("change-user", "Change User");
+  rootElement.appendChild(changeUserBtn);
+
+  const header = createHeader();
+  rootElement.appendChild(header);
+
+  const form = createForm();
+
+  const inputElement = createInput();
+  form.appendChild(inputElement);
+
+  const addButton = createBtn("add", "Add");
+  form.appendChild(addButton);
+
+  const filterOption = createFilterElement();
+  form.appendChild(filterOption);
+
+  rootElement.appendChild(form);
+
+  const listHeader = createHeading("h2", "list-header", "To Do List for");
+  rootElement.appendChild(listHeader);
+
+  const todoList = createTodoList();
+  rootElement.appendChild(todoList);
+
+  // global variables
+  let userName;
+  let todoItems = [];
```

Comment:
these are not global variables anymore as they are inside the `createApp` function's scope (comment can be removed)